dapp_manager/dapp_manager.py

```diff
@@ -88,6 +92,40 @@ def read_file(self, file_type: RunnerFileType, ensure_alive: bool = True) -> str
             self._ensure_alive()
         return self.storage.read_file(file_type)
 
+    def exec_command(self, service: str, command: List[str], timeout: int):
+        self._ensure_alive()
+        m = re.match("^(?P<service_name>.*?)(\\[(?P<idx>\\d+)\\])?$", service)
+        if not m:
+            raise ValueError("`service` parameter format unknown, use ")
+        service_name = m.group("service_name")
+        service_idx = int(m.group("idx") or 0)
+        command_msg = json.dumps({service_name: {service_idx: command}})
+        start = datetime.now()
+        with self.storage.open("data", "r") as data:
+            data.seek(0, 2)
+            self.storage.write_file("commands", command_msg)
```

Comment:
I think this `self.storage.write_file("commands", command_msg)` can be moved to line 103
```python
102        command_msg = json.dumps({service_name: {service_idx: command}})
103        self.storage.write_file("commands", command_msg)
```

Reply:
How about 124
```python
123        command_msg = json.dumps({service_name: {service_idx: command}})
124        self.storage.write_file("commands", command_msg)
```